aten/src/ATen/test/vmap_test.cpp

```diff
@@ -850,4 +850,217 @@ TEST(VmapTest, TestBatchedTensorPermute) {
   }
 }
 
+static void checkMultiBatchVmapTransform(TensorList inputs, TensorList expected_outputs) {
+  auto outputs = MultiBatchVmapTransform::logicalToPhysical(inputs);
+  ASSERT_EQ(outputs.size(), expected_outputs.size());
+  for (int64_t idx = 0; idx < outputs.size(); idx++) {
+    const auto& output = outputs[idx].tensor();
+    ASSERT_EQ(output.data_ptr(), expected_outputs[idx].data_ptr());
+    ASSERT_EQ(output.sizes(), expected_outputs[idx].sizes());
+    ASSERT_TRUE(at::allclose(output, expected_outputs[idx]));
+  }
+}
+
+TEST(VmapTest, TestMultiBatchVmapTransformBatchedBatched) {
+  {
+    // Check that batch dims get moved to the front
+    int64_t B0 = 5, B1 = 7;
+    Tensor x = at::randn({2, B0, 3, B1});
+    Tensor y = at::randn({B1, 2, 3, B0});
+    Tensor batched_x = makeBatched(x, {{0, 1}, {1, 3}});
+    Tensor batched_y = makeBatched(y, {{0, 3}, {1, 0}});
+
+    checkMultiBatchVmapTransform(
+        {batched_x, batched_y},
+        {x.permute({1, 3, 0, 2}), y.permute({3, 0, 1, 2})});
```

Comment:
NGL, probably would have been easier if we just wrote `B0, B1, 2, 3` as the expected result (and computed the permutation implicitly)

Reply:
This is where that `torch.moveaxis` PR comes in handy. I've changed the instances of permute introduced in this PR to use `at::movedim`, which is easier to read IMO than permute.